client/webpack.config.js

```diff
@@ -0,0 +1,263 @@
+    loaders: [
+      path.join(CLIENT_PATH, './webpack/web_loaders/globals-loader.js?clientPath=' + CLIENT_PATH),
+      'coffee'
+    ]
+  }
+])
+
+// JS, JSON, coffee
+pushLoader([
+  {
+    test: /\.js$/,
+    loaders: ['babel'],
+    // this might cause some problems.
+    exclude: [
+      /node_modules/,
+    ].concat(
+      // we are also excluding the test paths from third party libraries.
+      thirdPartyLoaders.map(function (loader) {
+        return loader.test
+      })
+    )
+  },
+  {
+    test: /\.coffee$/,
+    include: __dirname,
+    exclude: [
+      path.join(__dirname, 'src'),
+      // globals has its own loader
+      require.resolve('./globals.coffee'),
+    ],
+    loaders: [
+      'pistachio', 'coffee', 'cjsx'
+    ],
+  },
+  {
+    test: /\.json$/,
+    loaders: ['json'],
+    include: __dirname,
+  },
+])
+
+// Style loaders configuration
+pushLoader([
+  {
+    test: /\.styl$/,
+    include: CLIENT_PATH,
+    loaders: [
+      'style', 'css', 'stylus'
+    ]
+  },
+  {
+    test: /\.css$/,
+    include: CLIENT_PATH,
+    loaders: [
+      'style', 'css'
+    ],
+  },
+])
+
+// File & Url loaders
+pushLoader([
+  {
+    test: /\.(png|jpg|gif|woff|otf)/,
+    loader: 'url',
+    query: {
+      limit: 8192,
+      // this name will be appended to webpackConfig.output.publicPath
+      // if file size is greater than 8kb.
+      name: '[path][name].[ext]'
+    }
+  },
+  {
+    test: /\.ttf$/,
+    loader: 'file',
+    query: {
+      mimetype: 'application/x-font-ttf'
+    }
+  },
+  {
+    test: /\.eot$/,
+    loader: 'file',
+    query: {
+      mimetype: 'application/octet-stream'
+    }
+  },
+  {
+    test: /\.svg$/,
+    loader: 'file',
+    query: {
+      mimetype: 'image/svg+xml'
+    }
+  },
+])
+
+// plugins
+
+webpackConfig.plugins = [
+  // move thirdparty folder
+  new CopyWebpackPlugin([
+    { from: THIRD_PARTY_PATH, to: path.join(BUILD_PATH, '..', 'thirdparty') }
+  ]),
+  new ProgressBarPlugin({
+    format: ' client: [:bar] :percent ',
+    // to trigger 100% terminal width.
+    width: 1024
+  })
+]
+
+// development environment specific plugins.
+if (__DEV__) {
+  webpackConfig.plugins.push(
+    new WebpackNotifierPlugin({ title: 'Koding Frontend' })
+  )
+}
+// prod environment specific plugins.
+else if (__PROD__) {
+  webpackConfig.plugins.push(
+    new webpack.optimize.OccurrenceOrderPlugin(),
+    new webpack.optimize.DedupePlugin(),
+    new webpack.optimize.UglifyJsPlugin({
+      compress: {
+        unused: true,
+        dead_code: true,
+        warnings: false,
+      }
+    })
+  )
+}
+
+// Configure stylus that way we can require styl files and the final bundle
+// will include a css bundle as well.
+webpackConfig.stylus = {
+  use: [require('nib')()],
+  import: [
+    '~nib/lib/nib/index.styl',
+    COMMON_STYLES_PATH
+  ],
+  define: {
+    assetsPath: '/assets',
+    rootPath: CLIENT_PATH,
+  }
+}
+
+module.exports = webpackConfig
```

Comment:
Hello @burakcan Thank you so much for the suggestion.

I was planning to do the same, i decided not to, because at this point differences are so small that creating other config files just because we _may_ need it in the future would be an overkill.